elastic/kibana
# [Metrics UI] Add ability for user to set anomaly threshold

## Summary

Resolves https://github.com/elastic/kibana/issues/88548
Adds the ability for user to set anomaly threshold.
<img width="1025" alt="Screen Shot 2021-02-05 at 8 10 17 AM" src="https://user-images.githubusercontent.com/1676003/107037845-acaafe80-6789-11eb-9d00-658e765d6ccb.png">


## Test
- Generate low severity anomaly [using slingshot](https://gist.github.com/Zacqary/22ff439ccdf9ba748e9ff8fef1ba8799) in timeline
    - with a fresh ES, run `node examples/pod-script.js` with default settings to generate historical data or connect to the observability clusters
    - Go to Settings tab, scroll to the Machine Learning panel, and set the severity level to being very low, say 5
    - Go to the Metrics app and enable jobs for Kubernetes Pods
    - Go back to the Inventory tab and select Show Kubernetes Pods with Metrics set to memory
    - Open the timeline and you should see anomalies.  When you go to the anomalies page you can see the severity level is >= to the level you set
   
Eg 1 of low severity anomaly showing in timeline:
<img width="527" alt="Screen Shot 2021-02-05 at 2 18 29 PM" src="https://user-images.githubusercontent.com/1676003/107079297-a20a5c80-67bd-11eb-94d2-c09ebad5ff20.png">
<img width="794" alt="Screen Shot 2021-02-05 at 2 16 37 PM" src="https://user-images.githubusercontent.com/1676003/107079304-a59de380-67bd-11eb-9b88-7a3f9a5bd1fc.png">
Eg 2 of low severity anomaly showing in timeline:
<img width="381" alt="Screen Shot 2021-02-05 at 2 42 48 PM" src="https://user-images.githubusercontent.com/1676003/107081406-a126fa00-67c0-11eb-912b-320fd4eebf2a.png">
<img width="1222" alt="Screen Shot 2021-02-05 at 2 42 04 PM" src="https://user-images.githubusercontent.com/1676003/107081470-bbf96e80-67c0-11eb-898c-8a46b2990773.png">
    - You can also view that the requests for anomalies in both hosts and k8s will send the new `anomalyThreshold` parameter in the request and it should match what you've set
  ` /api/infra/infra_ml/results/metrics_k8s_anomalies` or `/api/infra/infra_ml/results/metrics_hosts_anomalies`:
<img width="1495" alt="Screen Shot 2021-02-05 at 2 48 36 PM" src="https://user-images.githubusercontent.com/1676003/107081808-4215b500-67c1-11eb-9e0f-f64d3a653607.png">
- The panel in settings should only be shown if the user has ML capabilities

### Checklist

Delete any items that are not applicable to this PR.
